randovania/generator/filler/retcon.py

```diff
@@ -175,12 +183,15 @@ def select_weighted_action(rng: Random, weighted_actions: Mapping[Action, float]
         return rng.choice(list(weighted_actions.keys()))
 
 
-def increment_considered_count(locations_weighted: WeightedLocations) -> None:
+def increment_index_age(locations_weighted: WeightedLocations, increment: float) -> None:
+    """
+    Increments the pickup index's age for every already collected index.
+    """
     for player, location, _ in locations_weighted.all_items():
-        was_present = location in player.pickup_index_considered_count
-        player.pickup_index_considered_count[location] += 1
+        was_present = location in player.pickup_index_age
+        player.pickup_index_age[location] += increment
```

Comment:
Did you consider not incrementing the age at all when there are no new pickups?

Reply:
That function is only called when assigning a pickup

Reply:
I was under the impression that the challenge is that when the generator has to place 5 missile expansions to unlock Tower of Light, then when assigning the first 4 expansions, there are no new locations being unlocked, but all open locations get their age incremented every time, so after assigning the fifth one, then with Tower of Light being unlocked, that leaves every other location with a comparatively much higher age compared to when most items are assigned.

Further, when there are multiple missile requirements in the world, with different numbers of required missiles, then it's going to be hard to balance the location weights based on assigning a value to the item type.

Reply:
Ah, the issue is that discovering there are no new pickups unlocked comes much later. In particular, in the next time this world is considered for assigning a pickup.

Not impossible to do, but it'll be a significant redesign.

Reply:
I see. Another angle would be whether there is some information available when deciding to place a missile expansion, or any other item, about whether that pickup by itself opens anything, or if it already knows that it will need more of them.